### PR TITLE
Yatomata OSGified

### PR DESCRIPTION
**Pull request by skapral**
Made Yatomata usable in OSGi environment by generating necessary Manifest headers. Additionally made several classes, which (I believe) not belong to public API, hidden.

### CODE REVIEW

**Comment by smecsia** [comment]
@skapral thank you! This looks good :+1:

File: src/main/java/ru/yandex/qatools/fsm/impl/YatomataImpl.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
 /**
  * @author Ilya Sadykov
  */
-public class YatomataImpl<T> implements Yatomata<T> {
+class YatomataImpl<T> implements Yatomata<T> {
```
Comment (vania-pooh) [inline src/main/java/ru/yandex/qatools/fsm/impl/YatomataImpl.java]: What is the expected way of using Yatomata classes with Spring (i.e. when FSM class has autowired fields)? This change breaks my current implementation: https://github.com/meridor/perspective-backend/blob/master/perspective-worker/src/main/java/org/meridor/perspective/worker/processor/impl/FSMBuilderAwareImpl.java
The use case of the class above is:
https://github.com/meridor/perspective-backend/blob/master/perspective-worker/src/main/java/org/meridor/perspective/worker/processor/ImagesProcessor.java#L50

Comment (skapral) [inline src/main/java/ru/yandex/qatools/fsm/impl/YatomataImpl.java]: Sorry for inconvenience. If this class is actually a part of a public contract, I'll make another request to turn it back.

Comment (vania-pooh) [inline src/main/java/ru/yandex/qatools/fsm/impl/YatomataImpl.java]: @smecsia: probably there's better way to get Yatomata instance than mine when using Spring. Is there?

Comment (smecsia) [inline src/main/java/ru/yandex/qatools/fsm/impl/YatomataImpl.java]: @vania-pooh we probably have to add the ability to instantiate the FSM with the reference FSM instance from the FSMBuilder. Impl classes usually have to be non-public. 
Can you send a separate PR for that?

Comment (vania-pooh) [inline src/main/java/ru/yandex/qatools/fsm/impl/YatomataImpl.java]: Sure. Will do.
